tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/contact/OpenPaasContactsConsumer.java

```diff
@@ -117,16 +130,32 @@ private Mono<EmailAddressContact> messageConsume(AcknowledgableDelivery ackDeliv
             });
     }
 
-    private Mono<EmailAddressContact> handleMessage(ContactAddedRabbitMqMessage contactAddedMessage) {
-        LOGGER.trace("Consumed jCard object message: {}", contactAddedMessage);
-        Optional<ContactFields> maybeContactFields = contactAddedMessage.vcard().asContactFields();
+    private Mono<?> handleMessage(ContactRabbitMqMessage contactMessage, String queue) {
+        LOGGER.trace("Consumed jCard object message: {}", contactMessage);
+        Optional<ContactFields> maybeContactFields = contactMessage.vcard().asContactFields();
+        ContactHandler contactHandler = contactHandlers.get(queue);
+        if (contactHandlers.get(queue) == null) {
```

Comment:
We should probably move this at the beginning of the method.

Reply:
And change the exception message message to something more descriptive. LLMs can come handy in this case.

Reply:
The case of being asked to hanlde a queue  that do not exist should not even be a thing.
We should by design always cary on the information of what to do without using side bands to pass the information.